### PR TITLE
Added Redeem entities for origin and auxiliary subgraph

### LINKED ISSUE #118: Generate subgraph for events related to redeem flow
Add support of below entities in subgraph to enable redeem for facilitator. 
1. RedeemIntentDeclared
2. RedeemProgressed
3. RedeemIntentConfirmed
4. UnstakeProgressed

### LINKED ISSUE #125: Added RedeemPool contract subgraph entities in auxiliary subgraph
RedeemPool contract is implemented in https://github.com/mosaicdao/mosaic-contracts/pull/790. 

Auxiliary subgraph should also track event for RedeemPool in order to enable redeem and unstake.

Implementation of this ticket should add RedeemPool events to auxiliary subgraph which is deployed when auxiliary chain starts.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Fixes #118 

Fixes #125 

This ticket adds redeem entities in the subgraph.

### CODE REVIEW

**Comment by deepesh-kn** [comment]
Let's hold this PR till https://github.com/mosaicdao/mosaic-contracts/pull/790 gets merged.

**Comment by abhayks1** [comment]
Started reviewing...

File: graph/origin/subgraph.yaml.mustache [hunk 1]
```diff
@@ -43,5 +43,29 @@ dataSources:
           handler: handleStakeIntentDeclared
         - event: StakeProgressed(bytes32,address,uint256,uint256,bool,bytes32)
           handler: handleStakeProgressed
+        - event: RedeemIntentConfirmed(bytes32,address,uint256,address,uint256,uint256,bytes32)
```
Comment (abhayks1) [inline graph/origin/subgraph.yaml.mustache]: entities of redeem flow are missing 🔴

File: graph/origin/subgraph.yaml.mustache [hunk 2]
```diff
@@ -43,5 +43,29 @@ dataSources:
...
+          handler: handleRedeemIntentConfirmed
```
Comment (abhayks1) [inline graph/origin/subgraph.yaml.mustache]: GatewayProven entity and handler is missing ⏰

**Pull request merged**